nixos/modules/services/misc/matrix-corporal.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, pkgs, config, ... }:
+let
+  cfg = config.services.matrix-corporal;
+in
+{
+  options.services.matrix-corporal = {
```

Comment:
You should add some options (like `authSharedSecretFile`) for secrets so they aren't placed in `settings` and end up in the nix store.
If you do this you can have a `preStart` script to replace the secrets with `jq` at runtime and keep your secrets safe.

Let me know if you need any details on this.

Reply:
oh, hmm that is pretty tricky, especially the `AuthorizationBearerToken` stuff that _can_ be part of the config file but aren't necessarily.

I'm actually not sure how to deal with that in a sensible way

Reply:
Make the option `type = with types; nullOr path;` and set `default = null;`. From here you add to the `systemd` service:
```
  systemd.services.matrix-corporal = {
    ...
    preStart = optionalString (cfg.authorizationBearerTokenFile) ''
      use jq to insert AuthorizationBearerToken: 'cat ${cfg.authorizationBearerTokenFile}' into a new copy of the file under /var/lib/matrix-corporal
    '';
  };
```

Or some variation of that.